client/login/wp-login/index.jsx

```diff
@@ -355,9 +355,10 @@ export class Login extends Component {
 		}
 
 		// use '?signup_url' if explicitly passed as URL query param
-		const signupUrl = this.props.signupUrl
+		const signup_url_en = this.props.signupUrl
 			? window.location.origin + pathWithLeadingSlash( this.props.signupUrl )
 			: getSignupUrl( query, currentRoute, oauth2Client, locale, pathname );
+		const signupUrl = addLocaleToPath( signup_url_en, locale );
```

Comment:
It appears that the `locale` parameter is already passed to the `getSignupUrl` function call above.

At first glance, it seems like `getSignupUrl` handles [quite a few use cases](https://github.com/Automattic/wp-calypso/blob/trunk/client/lib/login/index.js#L47), so I don't think forcefully appending the locale is a good solution here.

I'd suggest to look into the `getSignupUrl` function instead and troubleshoot why it doesn't take into account the `locale` parameter that's already being passed to it.